### DISCUSSION TITLE
in nextJs with useSearchParams(), fancyapps fancybox can open but close immediately

### DISCUSSION THREAD

**Discussion by TheCabinDev** [fancyapps/ui#676] [General]
what is the best way to use useSearchParams() in nextJs, with fancybox. 

my problem is when i comment useSearchParams(), the fancybox gallery work normally. but when use useSearchParams(), the gallery will open and immediatly close

**Comment by alvinChristianto** [comment]
same here, not yet have solutions

**Comment by fancyapps** [comment]
If you are using example React code from the docs, then try commenting out
the closing on component rerender.

**Reply by TheCabinDev** [reply]
Yes I follow the docs : https://fancyapps.com/resources/integration/#react , but where to commenting the components rerender ?

**Reply by TheCabinDev** [reply]
I did comment : NativeFancybox.close(); and it works. Any infos I need to know @fancyapps ?

**Reply by fancyapps** [reply]
To be honest, I don't know what additional info you would like to hear. Basically, you can choose to close Fancybox or not when re-rendering the component, it depends on your preferences/requirements. If you're an experienced React (or Vue, Svelte, etc.) developer, you can easily create your own component, because Fancybox provides an API to easily work with Fancybox instances.
